src/jsonrpc-remote-machine.cpp

```diff
@@ -576,137 +722,113 @@ static json jsonrpc_get_version_handler(const json &j, mg_connection *con, http_
+        // Note that the parent doesn't need the server that will be used by the child,
+        // we can close it.
+        beast::error_code ec;
+        acceptor.close(ec);
+        SLOG(trace) << session->handler->local_endpoint << " fork parent";
+    } else { // parent and fork() failed
+        const int errno_copy = errno;
+        SLOG(error) << session->handler->local_endpoint << " fork failed (" << strerror(errno_copy) << ")";
         return jsonrpc_response_server_error(j, "fork failed ("s + strerror(errno_copy) + ")"s);
     }
-    if (ret == 0) { // child
-        // The child doesn't need the event_manager it inherited from the parent.
-        // However, it would be impolite to destroy it here, since it would be somewhat unexpected (reentrant?).
-        // I.e., we are currently in a function that was invoked from within mg_mgr_poll.
-        // So we return with a status and destroy the event_manager only after mg_mgr_poll returns in our loop.
-        h->status = http_handler_status::forked_child;
-        h->child->server_address = new_server_address;
-        h->child->machine = std::move(h->machine);
-        return json{};
-    }
-    // parent
-    // The parent doesn't need the server that will be used by the child
-    mg_mgr_free_ours(&h->child->event_manager);
-    delete h->child;
-    h->child = nullptr;
-    con->is_draining = 1;
     return jsonrpc_response_ok(j, new_server_address);
 }
 
 /// \brief JSONRPC handler for the rebind method
 /// \param j JSON request object
-/// \param con Mongoose connection
-/// \param h Handler data
+/// \param session HTTP session
 /// \returns JSON response object
 /// \details Changes the address the server is listening to.
 /// After this call, all new connections should be established using the new server address.
-static json jsonrpc_rebind_handler(const json &j, mg_connection *con, http_handler_data *h) {
+static json jsonrpc_rebind_handler(const json &j, const std::shared_ptr<http_session> &session) {
     static const char *param_name[] = {"address"};
     auto args = parse_args<std::string>(j, param_name);
     const std::string new_server_address = std::get<0>(args);
-    // Listen in the new port
-    auto *new_listen_connection = mg_http_listen(&h->event_manager, new_server_address.c_str(), http_handler, h);
-    if (!new_listen_connection) {
-        return jsonrpc_response_server_error(j, "rebind failed listening on "s + new_server_address);
-    }
-    // Mark connection to be drained
-    con->is_draining = 1;
-    // Mark previous listen connection to be closed
-    h->listen_connection->is_closing = 1;
-    // Set the new listen connection
-    h->server_address = new_server_address;
-    h->listen_connection = new_listen_connection;
+    const tcp::endpoint new_local_endpoint = address_to_endpoint(new_server_address);
+    if (new_local_endpoint.port() == 0) {
+        throw std::runtime_error{"rebind cannot be performed on port 0"};
```

Comment:
Don't see why yes but neither why not.

Reply:
Because the `rebind` request doesn't return the new endpoint to connect to, the client would not know the new address to connect to, so it wouldn't make sense to allow it to perform.